interfaces/builtin/unity8.go

```diff
@@ -0,0 +1,182 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/dbus"
+	"github.com/snapcore/snapd/snap"
+)
+
+var unity8ConnectedPlugAppArmor = []byte(`
+# Description: Can access unity8 desktop services
+
+#include <abstractions/dbus-session-strict>
+
+# Fonts
+#include <abstractions/fonts>
+/var/cache/fontconfig/   r,
+/var/cache/fontconfig/** mr,
+
+# The snapcraft desktop part may look for schema files in various locations, so
+# allow reading system installed schemas.
+/usr/share/glib*/schemas/{,*}              r,
+/usr/share/gnome/glib*/schemas/{,*}        r,
+/usr/share/ubuntu/glib*/schemas/{,*}       r,
+
+# URL dispatcher. All apps can call this since:
+# a) the dispatched application is launched out of process and not
+#    controllable except via the specified URL
+# b) the list of url types is strictly controlled
+# c) the dispatched application will launch in the foreground over the
+#    confined app
+dbus (send)
+     bus=session
+     path=/com/canonical/URLDispatcher
+     interface=com.canonical.URLDispatcher
+     member=DispatchURL
+     peer=(name=com.canonical.URLDispatcher,label=###SLOT_SECURITY_TAGS###),
+
+# This is needed when the app is already running and needs to be passed in
+# a URL to open. This is most often used with content-hub providers and
+# url-dispatcher, but is actually supported by Qt generally.
+dbus (receive)
+     bus=session
+     path=/###PLUG_DBUS_APPIDS###
+     interface=org.freedesktop.Application
+     member=Open
+     peer=(label=###SLOT_SECURITY_TAGS###),
+
+# Unity launcher (e.g. app counter, progress, alert)
+dbus (receive, send)
+     bus=session
+     path=/com/canonical/Unity/Launcher/###PLUG_DBUS_APPIDS###
+     peer=(name=com.canonical.Unity.Launcher,label=###SLOT_SECURITY_TAGS###),
+
+# Note: content-hub may become its own interface, but for now include it here
+# Pasteboard via Content Hub
+dbus (send)
+     bus=session
+     interface=com.ubuntu.content.dbus.Service
+     path=/
+     member={CreatePaste,GetAllPasteIds,GetLatestPasteData,GetPasteData,GetPasteSource,PasteFormats,RequestPasteByAppId,SelectPasteForAppId,SelectPasteForAppIdCancelled}
+     peer=(name=com.ubuntu.content.dbus.Service,label=###SLOT_SECURITY_TAGS###),
+dbus (receive)
+     bus=session
+     interface=com.ubuntu.content.dbus.Service
+     path=/
+     member={PasteboardChanged,PasteFormatsChanged,PasteSelected,PasteSelectionCancelled}
+     peer=(name=com.ubuntu.content.dbus.Service,label=###SLOT_SECURITY_TAGS###),
+
+# Lttng tracing is very noisy and should not be allowed by confined apps.
+# Can safely deny. LP: #1260491
+deny /{dev,run,var/run}/shm/lttng-ust-* r,
+`)
+
+var unity8ConnectedPlugSecComp = []byte(`
+recv
+recvfrom
+recvmsg
+send
+sendmsg
+sendto
```

Comment:
Please merge with trunk then you can drop recv* and send* (leaving just shutdown).